HighwayPatrol.cpp

```diff
@@ -22,12 +27,26 @@ void HighwayPatrol::scanHighway(Highway* h)
 
 void HighwayPatrol::pullOver( Vehicle* v, bool willArrest, Highway* h )
 {
+    std::string vehicleType;
     std::cout << "\n\n";
     std::cout << name << ": vehicle is traveling " << v->speed - h->speedLimit << " miles per hour over the speed limit" << std::endl;
     if( willArrest )
     {
         //print the vehicle type in this std::cout between "THE [" and "] PULL". 
-        std::cout << name << ": YOU IN THE [ " << " ] PULL OVER AND SHOW YOUR HANDS" << std::endl;
+        if( v == dynamic_cast<Car*>(v))
+        {
+            vehicleType = "Car";
+        }
+        else if( v == dynamic_cast<Motorcycle*>(v))
+        {
+            vehicleType = "Motorcycle";
+        }
+        else if( v == dynamic_cast<SemiTruck*>(v))
+        {
+            vehicleType = "Semi Truck";
+        }
```

Comment:
you don't need `v` here.  the result of the cast will either be a pointer or a `nullptr`.  that's equivalent to `true` or false`.
- [ ] simplify to `if( dynamic_cast<...` for each `if()`